### PR TITLE
Cycle hire chp8

### PR DESCRIPTION
**Pull request by KiranmayiV**
Hi @Robinlovelace, in lines 907-986 I have included the code for the shiny app which works when I run the code chunk however I haven't tested it rendering it with bookdown. The original coffee app in that code chunk seems to have been deployed and called into the markdown, please feel free to do the same with the CycleHireApp as well. (In this case the code for the app can be removed from the chunk).

In regard to the shiny app's code, I included 'observe' and 'leafletProxy' functions to make it consistent with the text. In the original code I shared. these were not included. I made some changes in the text to make it consistent with the new app. Hope they are ok.

Please let me know if there are any further changes needed, would be happy to edit.

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Many thanks @KiranmayiV I will aim to take a look and provide comments later this evening.

**Comment by Robinlovelace** [comment]
Or this evening... Apologies for the delay.

File: 08-mapping.Rmd [hunk 1]
```diff
@@ -905,12 +905,89 @@ One potential problem with the ease of developing prototypes with **shiny** is t
 For that reason, despite advocating **shiny**, we recommend starting with the longer established technology of a pen and paper as the first stage for interactive mapping projects.
 This way your prototype web applications should be limited not by technical considerations, but by your motivations and imagination.
 
-```{r coffeeApp-html, echo=FALSE, message=FALSE, fig.cap="coffeeApp, a simple web mapping application for exploring global coffee production in 2016 and 2017. Interactive version available online at geocompr.robinlovelace.net.", fig.scap="coffeeApp, a simple web mapping application.", eval=knitr::is_html_output()}
-knitr::include_app("https://shiny.robinlovelace.net/coffeeApp/")
+```{r CycleHireApp-html, echo=FALSE, fig.cap="Hire a cycle App, message=FALSE," a simple web mapping application for finding the closest cycle hiring station based on your location and requirement of cycles. Interactive version available online at geocompr.robinlovelace.net.", fig.scap="Cycle Hire App, a simple web mapping application.", eval=knitr::is_html_output()}
+
```
Comment (Robinlovelace) [inline 08-mapping.Rmd:909]: Heads-up @KiranmayiV the code here is causing the CI checks to fail. Can you remove these lines of code?

File: CycleHireApp/app.R [hunk 2]
```diff
@@ -0,0 +1,76 @@
+#Shiny app for cycle hire from https://github.com/Robinlovelace/geocompr/issues/584
+# Author - Kiranmayi Vadlamudi
+#25th Dec 2020
+
+library(pacman)
+p_load(sf, shiny, spData, leaflet, tidyverse, spDataLarge, units)
```
Comment (Robinlovelace) [inline CycleHireApp/app.R:6]: Could you replace these lines with something that doesn't depend on pacman? It's a great package but don't want to add extra dependencies. The following lines should do the same thing so could make a good replacement:

```r
pkgs = c("sf", "shiny", "spData", "leaflet", "tidyverse", "spDataLarge", "units")
invisible(lapply(pkgs, library, character.only = TRUE))

**Review by Robinlovelace — CHANGES_REQUESTED** [review]
Thanks for the PR, looking good. I've requested a few changes. If you can make those, which should make the GitHub Actions checks pass, I think we can merge this. Any questions you have on this, feel free to ask, really like the concept of the app.

File: 08-mapping.Rmd (hunk 1 above)
Comment (KiranmayiV) [reply 08-mapping.Rmd:909]: yup, done.

File: CycleHireApp/app.R (hunk 2 above)
Comment (KiranmayiV) [reply CycleHireApp/app.R]: Done.

File: 08-mapping.Rmd (hunk 1 above)
Comment (KiranmayiV) [reply 08-mapping.Rmd:909]: sorry my bad, there was a error in the chunk's caption. I corrected it. However the check are still failing.

**Review by Robinlovelace — APPROVED** [review]
Great work @KiranmayiV, most of the tests are passing and the updates look good. :+1: from me.

**Comment by Robinlovelace** [comment]
One question @KiranmayiV if we think of ways to improve the app later are you happy for us to do so?

**Comment by KiranmayiV** [comment]
> One question @KiranmayiV if we think of ways to improve the app later are you happy for us to do so?

Please feel free to improve it. I am also happy to help out.

**Comment by Robinlovelace** [comment]
> Please feel free to improve it. I am also happy to help out.

Great, I can think of a couple of possible improvements but will merge first and see how it works out!

File: 08-mapping.Rmd (hunk 1 above)
[the remark this answers is not in the document]
Comment (KiranmayiV) [reply 08-mapping.Rmd:909]: Hi @Robinlovelace , just noticed right now the text mentions CycleHireApp, but the app displayed is still coffee app, it needs to be updated in the code above with your URL.

**Pull request merged**